### ISSUE TITLE
imgui on touchscreen

### ISSUE THREAD

**Issue by sailfish009** [ocornut/imgui#2650]
Hi, not much to add my code here, because it is simple question.

**Version/Branch of Dear ImGui:**

Version: v1.49
Branch: master

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp
Compiler: vs2017
Operating System: Windows 10 64bit

**My Issue/Question:**

If i test button with mouse, it clicked with single-click.
But if i test button on touchscreen, it can not be clicked with single-click, 
but only double-click.

How i use imgui button on my touchscreen(microsoft surface, win10) with single-click?

**Screenshots/Video**

![realsense_viewer](https://user-images.githubusercontent.com/17582766/60762416-b2d97600-a04e-11e9-9dc6-9d10d4772942.png)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```

ImGui::Begin("Example Bug");
static int counter = 0;
if (ImGui::Button("-##1"))  // <----- single click doesn't work on touchscreen
{
    --counter;
}
ImGui::End();
```

Thanks!

**Comment by ocornut** [comment]
Forgot to fill requested issue template, closing. Please fill template and i’ll reopen.

**Comment by sailfish009** [comment]
> 
> 
> Forgot to fill requested issue template, closing. Please fill template and i’ll reopen.

Hi, i have updated my question.

**Comment by sailfish009** [comment]
[based ](https://stackoverflow.com/questions/45619710/need-to-tap-twice-on-touchscreen-to-click-button-in-wpf)on what i searched the problem, 

i need some converter single-click to double-click in imgui.

**Comment by bvgastel** [comment]
The problem is something else. In the Windows event layer, the touch down and touch up events are fired and processed in the same frame. So the event handling logic of ImGui does not encounter it. As a fix, if the touch up is in the same frame as touch down, handling of the touch up should be delayed until the next frame. I have some working code for this if you are still interested (also contains a fix for touch scrolling).

**Comment by rickyviking** [comment]
@bvgastel regarding your fix about scrolling, are you talking about scrolling with the lateral scrollbar on by touch+drag up/down the contents in the middle of the widget, as you would do on a phone?

**Comment by bvgastel** [comment]
In the middle of the widget, just like a phone.

**Comment by rickyviking** [comment]
@bvgastel nice! would you be able to share some code? Thank you

**Comment by ocornut** [comment]
> The problem is something else. In the Windows event layer, the touch down and touch up events are fired and processed in the same frame. So the event handling logic of ImGui does not encounter it. As a fix, if the touch up is in the same frame as touch down, handling of the touch up should be delayed until the next frame.

Closing this specific/old touch issue with should be solved by input trickling introduced in 1.87.
